walk-in-interview/src/main/java/com/google/account/business/servlets/CreateBusinessServlet.java

```diff
@@ -0,0 +1,114 @@
+        this.businessDatabase = new BusinessDatabase();
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            Optional<String> optionalUid = FirebaseAuthUtils.getUid(request);
+
+            if (!optionalUid.isPresent()) {
+                LOGGER.log(Level.SEVERE, /* msg= */"Illegal uid");
+                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                return;
+            }
+
+            String uid = optionalUid.get();
+
+            // Gets business object from the client
+            Business inputBusiness = parseBusinessAccount(request);
+
+            // Fetches the existing business object from database
+            Optional<Business> existingBusinessOptional = this.businessDatabase.getBusinessAccount(uid).get();
+
+            Business business;
+
+            // Updates the preliminary business object at only the updatable fields
+            if (!existingBusinessOptional.isPresent()) {
+                business = Business.newBuilder()
+                        .setName(inputBusiness.getName())
+                        .setJobs(ImmutableList.of()).build();
+            } else {
+                business = Business.newBuilder()
+                        .setName(inputBusiness.getName())
+                        .build();
+            }
+
+            // Stores the account into cloud firestore
+            addBusinessAccount(uid, business);
+
+            // Sends the success status code in the response
+            response.setStatus(HttpServletResponse.SC_OK);
+        } catch (FirebaseAuthException | IOException | ServletException
+                | ExecutionException | TimeoutException | InterruptedException e) {
+            // TODO(issue/47): use custom exceptions
+            LOGGER.log(Level.SEVERE, /* msg= */"Error occur: " + e.getCause(), e);
+            // Sends the fail status code in the response
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        }
+    }
+
+    /** Parses into valid Business object from json received from client request. */
+    private Business parseBusinessAccount(HttpServletRequest request) throws IOException, IllegalArgumentException {
+        // Parses job object from the POST request
+        try (BufferedReader bufferedReader = request.getReader()) {
+            String businessAccountJsonStr =
+                    bufferedReader.lines().collect(Collectors.joining(System.lineSeparator())).trim();
+
+            if (StringUtils.isBlank(businessAccountJsonStr)) {
+                throw new IllegalArgumentException("Json for Business object is Empty");
+            }
+
+            Business rawBusiness = ServletUtils.parseFromJsonUsingGson(businessAccountJsonStr, Business.class);
+
+            // Validates the attributes via build()
+            return rawBusiness.toBuilder().build();
+        }
+    }
+
+    /** Adds the business account in the database. */
```

Comment:
change javadoc to reflect that existing business will be updated and only the fields that are set.